File: backend/app/schemas/ask.py
from typing import Any

from pydantic import BaseModel


class AskRequest(BaseModel):
    question: str
    conversation_id: str | None = None


class AskResponse(BaseModel):
    question: str
    route: str
    answer: str
    conversation_id: str
    data: list[dict[str, Any]] | None = None
    sources: list[dict[str, Any]] | None = None
    generated_sql: str | None = None
    
    

File: frontend/src/api/ask.ts
import client from './client'

export interface AskResponse {
  question: string
  route: string
  answer: string
  conversation_id: string
  data: Record<string, unknown>[] | null
  sources: { text: string; document_id: string; chunk_index: number; score: number }[] | null
  generated_sql: string | null
  query_id: string | null
}

export async function askQuestion(question: string, conversationId?: string) {
  const response = await client.post<AskResponse>('/ask', {
    question,
    conversation_id: conversationId ?? null,
  })
  return response.data
}
